engines/stark/resources/item.cpp

```diff
@@ -405,17 +405,14 @@ Common::Array<Common::Point> ItemVisual::listExitPositionsImpl() {
 	Common::Array<Common::Point> positions;
 	Common::Point invalidPosition(-1, -1);
 
-	Common::Array<PATTable *>::iterator element = pattables.begin();
-	while (element != pattables.end()) {
-		if ((*element)->getDefaultAction() == _exitAction) {
-			Common::Point hotspot = getAnim()->getHotspotPosition((*element)->getIndex());
+	for (uint i = 0; i < pattables.size(); ++i) {
+		if (pattables[i]->getDefaultAction() == _exitAction) {
+			Common::Point hotspot = getAnim()->getHotspotPosition(i);
```

Comment:
What is the usage of `Object::getIndex()` anyway? What's its difference with the index of a listed array like here?

Reply:
`Object::getIndex` is used to uniquely identify an entity in the list of children of the parent entity.

In this case, I don't know why the index in the array was used. It seems fragile to me.